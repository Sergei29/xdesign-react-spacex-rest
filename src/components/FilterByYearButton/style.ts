import styled from "styled-components";

export const SelectByYearContainer = styled.div`
  & > select {
    width: 100%;
  }
`;
