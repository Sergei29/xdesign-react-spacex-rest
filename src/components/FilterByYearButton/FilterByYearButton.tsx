import React from "react";
// style:
import { SelectByYearContainer } from "./style";

type Props = {
  arrYears: number[];
  funcHandleSelect: (intYear: number | null) => void;
  nIntSelectedYear: null | number;
};

/**
 * @description filter by year button
 * @param {Array} {arrYears list of years
 * @param {null|Number} nIntSelectedYear selected year
 * @param {Function} funcHandleSelect callback on selecting a new value }
 * @returns {JSX} markup, select input
 */
const FilterByYearButton: React.FC<Props> = ({
  arrYears,
  nIntSelectedYear,
  funcHandleSelect,
}) => {
  /**
   * @description select change handler
   * @param {Object} objEvent change event object
   * @returns {indefined} sets state
   */
  const funcHandleChange = (
    objEvent: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    const intValue = parseInt(objEvent.target.value);
    const nIntValue = intValue > 0 ? intValue : null;
    funcHandleSelect(nIntValue);
  };

  return (
    <SelectByYearContainer
      id="years"
      onChange={funcHandleChange}
      value={nIntSelectedYear ? nIntSelectedYear : "Filter by year"}
    >
      <option value={0}>Filter by year</option>
      {arrYears.map((intYear) => (
        <option key={intYear} value={intYear}>
          {intYear}
        </option>
      ))}
    </SelectByYearContainer>
  );
};

export default FilterByYearButton;
