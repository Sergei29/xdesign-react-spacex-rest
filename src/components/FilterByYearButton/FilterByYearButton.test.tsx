import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import FilterByYearButton from "./FilterByYearButton";

describe("FilterByYearButton", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const objMockProps = {
      arrYears: [2006, 2007, 2008, 2009],
      nIntSelectedYear: 2006,
      funcHandleSelect: jest.fn(),
    };
    wrapper = shallow(<FilterByYearButton {...objMockProps} />);
  });

  it("should render", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
