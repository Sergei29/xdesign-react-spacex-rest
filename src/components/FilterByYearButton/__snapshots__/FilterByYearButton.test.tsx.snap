// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByYearButton should match snapshot 1`] = `ShallowWrapper {}`;
